code/game/objects/structures/signs.dm

```diff
@@ -664,3 +664,46 @@
 	desc = "An unbelievably creepy cat clock that surveys the room with every tick and every tock."
 	icon = 'icons/obj/structures/props/catclock.dmi'
 	icon_state = "cat_clock_motion"
+
+//===================//
+//      Calendar     //
+//=================//
+
+/obj/structure/sign/calendar
+	name = "wall calendar"
+	desc = "Classic office decoration and a place to stare at maniacally."
+	icon_state = "calendar_civ"
+	var/calendar_faction
+
+/obj/structure/sign/calendar/get_examine_text(mob/user)
+	. = ..()
+	. += SPAN_INFO("The current date is: [time2text(world.realtime, "DDD, MMM DD")], [GLOB.game_year].")
+	if(length(GLOB.holidays))
+		. += SPAN_INFO("Events:")
+		for(var/holidayname in GLOB.holidays)
+			var/datum/holiday/holiday = GLOB.holidays[holidayname]
+			if(holiday.holiday_faction)
+				if(holiday.holiday_faction != calendar_faction)
+					continue
+			. += SPAN_INFO("[holiday.name]")
+			. += SPAN_BOLDNOTICE("[holiday.greet_text]")
+
+/obj/structure/sign/calendar/upp
+	icon_state = "calendar_upp"
+	desc = "Classic office decoration and a place to stare at maniacally, has a UPP logo on it, also all text is in Russian."
```

Comment:
`Classic office decor with a spot to stare at maniacally. Features a UPP logo, with all text in Russian.`

grammar should be fixed here

Reply:
fixed